src/linear_algebra.jl

```diff
@@ -187,3 +189,70 @@ function LinearAlgebra.norm(x::AbstractArray{Num}, p::Real=2)
         sum(x->abs(x)^p, x)^inv(p)
     end
 end
+
+"""
+    (a, b, islinear) = linear_expansion(t, x)
+
+When `islinear`, return `a` and `b` such that `a * x + b == t`.
+"""
```

Comment:
That's pretty cool! What about multiple variables?

Reply:
It will be doing the similar thing but will be `a1 * x1 + a2 * x2 + ... + b == t`.